libraries/Adafruit_LittleFS/src/Adafruit_LittleFS_File.cpp

```diff
@@ -46,9 +46,20 @@ File::File (Adafruit_LittleFS &fs)
 File::File (char const *filename, uint8_t mode, Adafruit_LittleFS &fs)
  : File(fs)
 {
+  // public constructor calls public API open(), which will obtain the mutex
   this->open(filename, mode);
 }
 
+File::File (uint8_t differentiator, char const *filename, uint8_t mode, Adafruit_LittleFS &fs)
+ : File(fs)
+{
+  // private constructor calls private API _open(), which does NOT obtain the mutex
+  (void)differentiator;
```

Comment:
could you tell me what is the `differentiator` for ?

Reply:
Good question.

This was a relic of a prior attempted solution, before `open()` was split into `open()` as a wrapper and `_open()` as the internal function without mutex.

<detail>
<summary>Background, in case you wonder...</summary>

Let me give some background on existing constructors and use:

* `File::File (Adafruit_LittleFS &fs)` -- This public constructor simply keeps a copy of the pointer.  No lock used.

* `File::File (char const *filename, uint8_t mode, Adafruit_LittleFS &fs) : File(fs)` -- This calls the above constructor, but then calls `open()`, which will obtain the mutex as part of the open().

At the time, I believe `openNextFile()` created the return value by using the second constructor (with the filepath + mode).  Therefore, it was necessary for the mutex to **NOT** be acquired to call that constructor, to avoid deadlock.  Obviously, this was not OK, so I added the private constructor with a differentiating argument, so it would only be callable from within the `File` class.

</detail>

But, as your sharp eyes noticed, this constructor is no longer necessary.  I've removed it now.

Reply:
Background, in case you wonder...

Let me give some background on existing constructors and use:

* `File::File (Adafruit_LittleFS &fs)` -- This public constructor simply keeps a copy of the pointer.  No lock used.

* `File::File (char const *filename, uint8_t mode, Adafruit_LittleFS &fs) : File(fs)` -- This calls the above constructor, but then calls `open()`, which will obtain the mutex as part of the open().

At the time, I believe `openNextFile()` created the return value by using the second constructor (with the filepath + mode).  Therefore, it was necessary for the mutex to **NOT** be acquired to call that constructor, to avoid deadlock.  Obviously, this was not OK, so I added the private constructor with a differentiating argument, so it would only be callable from within the `File` class.